packages/sanity/src/desk/components/breadcrumbs/Breadcrumbs.tsx

```diff
@@ -0,0 +1,101 @@
+import {Box, Text, useClickOutside, Popover, Stack} from '@sanity/ui'
+import React, {
+  Children,
+  Fragment,
+  forwardRef,
+  isValidElement,
+  useCallback,
+  useMemo,
+  useState,
+} from 'react'
+import {ChevronRightIcon} from '@sanity/icons'
+import {BreadcrumbItemRoot, ExpandButton, Root} from './breadcrumbs.styles'
+
+/**
+ * @internal
+ * @hidden
+ */
+export interface BreadcrumbsProps {
+  maxLength?: number
+}
+
+/**
+ * @internal
+ * @hidden
+ */
+export const Breadcrumbs = forwardRef(function Breadcrumbs(
+  props: BreadcrumbsProps & Omit<React.HTMLProps<HTMLOListElement>, 'as' | 'ref' | 'type'>,
+  ref: React.ForwardedRef<HTMLOListElement>
+) {
+  const {children, maxLength, ...restProps} = props
+  const [open, setOpen] = useState(false)
+  const [expandElement, setExpandElement] = useState<HTMLButtonElement | null>(null)
+  const [popoverElement, setPopoverElement] = useState<HTMLDivElement | null>(null)
+  const collapse = useCallback(() => setOpen(false), [])
+  const expand = useCallback(() => setOpen(true), [])
+
+  useClickOutside(collapse, [expandElement, popoverElement])
+
+  const rawItems = useMemo(
+    () =>
+      Children.toArray(children).filter((child) => {
+        return isValidElement(child)
+      }),
+    [children]
+  )
+
+  const items = useMemo(() => {
+    const len = rawItems.length
+
+    if (maxLength && len > maxLength) {
+      const afterLength = Math.floor(maxLength / 2)
+
+      return [
+        <Popover
+          constrainSize
+          content={
+            <Stack as="ol" overflow="auto" padding={2}>
+              {rawItems.slice(0, len - maxLength)}
+            </Stack>
+          }
+          key="button"
+          open={open}
+          placement="top"
+          portal
+          ref={setPopoverElement}
+        >
+          <ExpandButton
+            mode="bleed"
+            onClick={open ? collapse : expand}
+            padding={2}
+            ref={setExpandElement}
+            selected={open}
+            text="…"
+          />
+        </Popover>,
+        ...rawItems.slice(len - maxLength, len - afterLength),
+        ...rawItems.slice(len - afterLength),
```

Comment:
I have to admit I'm having a hard time following this section, is it to solely keep this at a specific number of breadcrumb items?

Reply:
Yes, this to support "maxItems" so that items move to a popover once it's over the max items. The logic is mostly taken from [sanity/ui](https://github.com/sanity-io/ui/blob/main/src/components/breadcrumbs/breadcrumbs.tsx#L41-L78) and modified a little bit to support the design case. I will add some comments to this piece to make more sense of it. Thanks for flagging it!

Reply:
I simplified it, hopefully it makes more sense now

Reply:
Think it's better! Thank you! :)